core/templates/components/summary-tile/topic-summary-tile.component.ts

```diff
@@ -46,6 +49,10 @@ export class TopicSummaryTileComponent {
           AppConstants.ENTITY_TYPE.TOPIC, this.topicSummary.getId(),
           this.topicSummary.getThumbnailFilename());
     }
+    this.topicNameTranslationKey = this.i18nLanguageCodeService.
+      getTopicTranslationKey(this.topicSummary.getId(), false);
+    this.topicNameTranslationKey = this.hasTranslationKey() ?
```

Comment:
Why are you overwriting the previous line here? Can you do the calculation without redefining this.topicNameTranslationKey?